src/Scenario/GeoSearchScenario.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TelegramOSINT\Scenario;
+
+use TelegramOSINT\Client\InfoObtainingClient\Models\GeoChannelModel;
+use TelegramOSINT\Exception\TGException;
+use TelegramOSINT\Logger\Logger;
+use TelegramOSINT\MTSerialization\AnonymousMessage;
+
+class GeoSearchScenario extends InfoClientScenario
+{
+    private const DECIMALS = 6;
+
+    /** @var array */
+    private $points;
+    /** @var callable|null */
+    private $onChatReady;
+    /** @var int */
+    private $limit;
+    /** @var int */
+    private $counter;
+
+    /**
+     * @param array                         $points
+     * @param callable|null                 $onReady         function(GeoChannelModel $model)
+     * @param ClientGeneratorInterface|null $clientGenerator
+     * @param int                           $limit
+     *
+     * @throws TGException
+     */
+    public function __construct(
+        array $points,
+        callable $onReady = null,
+        ClientGeneratorInterface $clientGenerator = null,
+        int $limit = 100
+    ) {
+        parent::__construct($clientGenerator);
+        $this->points = $points;
+        $this->onChatReady = $onReady;
+        $this->limit = $limit;
+        $this->counter = 0;
+    }
+
+    /**
+     * @param bool $pollAndTerminate
+     *
+     * @throws TGException
+     */
+    public function startActions(bool $pollAndTerminate = true): void
+    {
+        $this->login();
+        foreach($this->points as $point) {
+            [$lat, $lon] = $point;
+            if ($this->counter > $this->limit) {
+                break;
+            }
+            $this->infoClient->getLocated($lat, $lon, function (AnonymousMessage $message) use ($lat, $lon) {
+                /** @see https://core.telegram.org/constructor/updates */
+                if ($message->getType() != 'updates') {
+                    return;
+                }
+
+                foreach ($message->getNodes('chats') as $chat) {
```

Comment:
Wrapper class needed in order to not access by name directly

Reply:
Fixed, use wrapper now.